Kinvey/Kinvey/TaskProgressRequest.swift

```diff
@@ -125,7 +125,7 @@ class TaskProgressRequest: NSObject {
         {
             let progressStatus = ProgressStatus(task)
             if (progressStatus.countOfBytesSent == progressStatus.countOfBytesExpectedToSend && progressStatus.countOfBytesReceived >= 0 && progressStatus.countOfBytesExpectedToReceive != 0) ||
-                (progressStatus.countOfBytesSent >= 0 && progressStatus.countOfBytesExpectedToSend > 0)
+                (progressStatus.countOfBytesSent >= 0 && (progressStatus.countOfBytesExpectedToSend > 0 || progressStatus.countOfBytesExpectedToSend == -1))
```

Comment:
@heyzooi under what circumstances is `countOfBytesExpectedToSend` = 0 or -1 respectively? I suggest adding some comments to this code so it leaves some indication of when these cases are hit.